advanced/creating-a-plugin.md

```diff
@@ -1,43 +1,113 @@
+  // add plugin code here
+
+  return {
+    getMetadata: function () {
+      return {
+        name: 'Custom plugin',
+        url: 'https://example.com/docs/customplugin'
+      };
+    }
+  }
+});
 ```
-/tinymce/plugins/example/plugin.js
-/tinymce/plugins/example/plugin.min.js
+
+## Using custom plugins with {{site.productname}}
+
+Custom plugins can be used by a {{site.productname}} instance by either:
+
+- **Using `external_plugins`**: Use the [external_plugins option]({{ site.baseurl }}/configure/integration-and-setup/#external_plugins) to specify the URL based location of the entry point file for the plugin.
+- **Copy code into `plugins` folder**: Copy the entry point file (and any other files) into the `plugins` folder of the distributed {{site.productname}} code.
+
+The plugin can then be used by including it in the list of plugins specified by the [`plugins` option]({{ site.baseurl }}/configure/integration-and-setup/#plugins).
+
+> **Note:** {{ site.companyname }} recommends using the `external_plugins` option for custom plugins.
+
+### Example: Using `external_plugins`
+
+```js
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  external_plugins: {
+    pluginId: 'https://example.com/customplugincode.min.js'
+  },
+  plugins: ''
+});
 ```
 
-This new example plugin can be loaded using the `tinymce.init` plugins option.
+### Example: Copying plugin code into the `plugins` folder
 
 ```js
 tinymce.init({
-  selector: 'textarea',
-  plugins: 'example'
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'pluginId'
 });
 ```
 
-Developers can designate the location of the plugins by loading the `plugin.js` or `plugin.min.js` files directly after the `tinymce.js` or `tinymce.min.js`.
+## Testing {{site.productname}}
+
+Due to the amount of browser APIs used by {{site.productname}}, when testing {{site.productname}} or any custom plugins, you must use a testing framework that runs on a real browser. Testing frameworks that mock the browser will not work. 
+
+## Language localization
 
-## Example of loading the plugin from another URL
+{{site.productname}} comes with translations for many languages. To add additional strings to a supported language for a custom plugin, use the following procedure.
 
-```html
-<script src="/tinymce/js/tinymce.min.js"></script>
-<script src="/scripts/my.tinymce.plugin.js"></script>
+1. Create a "langs" directory in the plugin directory for custom translations. 
+2. For each language that the plugin supports, create a translation file. The files should be JavaScript files and use the relevant language code as the file name. For example, {{site.productname}} will look for a Spanish translation file at `'<your plugin>/langs/es_ES.js'`, where `<your plugin>` is to the directory that contains the plugin's entry point file. For a list of supported languages, see: [Supported languages]({{ site.baseurl }}/configure/localization/#supportedlanguages).
```

Comment:
```suggestion
2. For each language that the plugin supports, create a translation file.

    The files should be JavaScript files and use the relevant language code as the file name. For example: {{site.productname}} will search for a Spanish translation file at `'<your plugin>/langs/es_ES.js'`, where `<your plugin>` is to the directory that contains the plugin's entry point file. For a list of supported languages, see: [Supported languages]({{ site.baseurl }}/configure/localization/#supportedlanguages).
    
```
look = personification, search, less so.
Also, the indentation _shouldn't_ break the numbering make the para a child of the above list item...